File: October, 2020 Challenge/Minimum Depth of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def __init__(self):
        self.ans = math.inf
    
    def dfs(self, node, level):
        if node == None:
            return
        
        if node.left == None and node.right == None:
            self.ans = min(self.ans, level)
        
        self.dfs(node.left, level + 1)
        self.dfs(node.right, level + 1)
        
    def minDepth(self, root: TreeNode) -> int:
        if root == None:
            return 0
        
        self.dfs(root, 1)
        return self.ans
        

File: daily_challenge_during_phd/2023/July/Minimum Depth of Binary Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int mn;
    void traverse(TreeNode* root, int cnt) {
        if(root == NULL) return;

        if(root->left == NULL && root->right == NULL) {
            mn = min(mn, cnt);
            return;
        }

        traverse(root->left, cnt + 1);
        traverse(root->right, cnt + 1);
    } 
    int minDepth(TreeNode* root) {
        if(root == NULL) return 0;
        
        mn = INT_MAX;
        traverse(root, 1);

        return mn;
    }
};
